src/components/screens/LockScreen.tsx

```diff
@@ -0,0 +1,155 @@
+`;
+
+export const LockScreen: React.FC<LockScreenProps> = ({ navigation }) => {
+  const {
+    signIn,
+    isLoading: isSigningIn,
+    error,
+    authStatus,
+  } = useAuthenticationStore();
+  const [publicKey, setPublicKey] = useState<string | null>(null);
+  const [passwordValue, setPasswordValue] = useState("");
+  const { t } = useAppTranslation();
+
+  // Monitor auth status changes to navigate when unlocked
+  useEffect(() => {
+    if (authStatus === "AUTHENTICATED") {
+      // Add a small delay to ensure state is settled before navigation
+      const navigationTimeout = setTimeout(() => {
+        navigation.replace(ROOT_NAVIGATOR_ROUTES.MAIN_TAB_STACK);
+      }, 100);
+
+      return () => {
+        clearTimeout(navigationTimeout);
+      };
+    }
+    return undefined;
+  }, [authStatus, navigation]);
+
+  useEffect(() => {
+    const fetchActiveAccountPublicKey = async () => {
+      const retrievedPublicKey = await getActiveAccountPublicKey();
+      setPublicKey(retrievedPublicKey);
+    };
+
+    fetchActiveAccountPublicKey();
+  }, []);
+
+  const canContinue = useMemo(
+    () =>
+      passwordValue.length >= PASSWORD_MIN_LENGTH &&
+      passwordValue.length <= PASSWORD_MAX_LENGTH,
+    [passwordValue],
+  );
+
+  const handleUnlock = useCallback(() => {
+    if (!canContinue) return;
+
+    // Disable other navigation attempts while signing in
+    if (isSigningIn) return;
+
+    // Try to sign in - error handling is in the auth store
+    signIn({ password: passwordValue });
+    // Navigation will happen automatically through the authStatus effect
+  }, [canContinue, passwordValue, signIn, isSigningIn]);
+
+  const handlePasswordChange = useCallback((value: string) => {
+    setPasswordValue(value);
+  }, []);
+
+  return (
+    <BaseLayout useSafeArea useKeyboardAvoidingView>
+      <Container>
+        <StyledIconContainer>
+          <FreighterLogo width={px(48)} height={px(48)} />
+        </StyledIconContainer>
+        <StyledFormContainer>
+          <Avatar size="lg" publicAddress={publicKey ?? ""} />
+          <StyledTitle>{t("lockScreen.title")}</StyledTitle>
+          <Text secondary>{t("lockScreen.description")}</Text>
+          <StyledInputContainer>
+            <Input
+              isPassword
+              placeholder={t("lockScreen.passwordInputPlaceholder")}
+              fieldSize="lg"
+              value={passwordValue}
+              onChangeText={handlePasswordChange}
+            />
+            {error && <StyledErrorText>{error}</StyledErrorText>}
+            <Button
+              tertiary
+              lg
+              onPress={handleUnlock}
+              disabled={!canContinue}
+              isLoading={isSigningIn}
+            >
+              {t("lockScreen.unlockButtonText")}
+            </Button>
+          </StyledInputContainer>
+        </StyledFormContainer>
+        <ForgotPasswordContainer>
+          <Button secondary lg onPress={() => {}}>
```

Comment:
@Brunonascdev once this "Forgot Password" action is implemented will this take the user back to the Welcome Screen so that s/he can either import the wallet again using seed phrase or create a new one?

Is the user currently able to get back to the Welcome Screen if they logout and delete the app?

Asking just to understand how Teaser testers could get back to the Welcome Screen

Thanks!